fastestimator/dataset/batch_dataset.py

```diff
@@ -224,7 +232,11 @@ def __getitem__(self, batch_idx: int) -> List[Dict[str, Any]]:
                 num_samples = self.num_samples
             for dataset, num_sample, index_map in zip(self.datasets, num_samples, self.index_maps):
                 for idx in range(num_sample):
-                    items.append(dataset[index_map[batch_idx * num_sample + idx]])
+                    item = dataset[index_map[batch_idx * num_sample + idx]]
+                    if isinstance(item, list):
+                        items.extend(item)
+                    else:
+                        items.append(item)
         else:
             num_sample = self.num_samples[0]
```

Comment:
else branch doesn't need editing? Is the idea that disjoint keys have to have same size? wouldn't special handling still be needed if, for example, each disjoint dataset returned a list of 2 things?

Reply:
or actually, does the implementation of _check_input that you proposed just prevent the usage of these list outputs with disjoint ds? since there'd be more than 1 copy of a key?